moveit_ros/planning_interface/moveit_cpp/include/moveit/moveit_cpp/moveit_cpp.h

```diff
@@ -0,0 +1,1099 @@
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Ioan Sucan, Sachin Chitta, simon Goldstein */
+
+#ifndef MOVEIT_MOVEIT_CPP_MOVEIT_CPP_
+#define MOVEIT_MOVEIT_CPP_MOVEIT_CPP_
+
+#include <moveit/warehouse/constraints_storage.h>
+#include <moveit/macros/class_forward.h>
+#include <moveit/macros/deprecation.h>
+#include <moveit/planning_scene_monitor/current_state_monitor.h>
+#include <moveit/planning_scene_monitor/planning_scene_monitor.h>
+#include <moveit/robot_state/robot_state.h>
+#include <moveit_msgs/RobotTrajectory.h>
+#include <moveit_msgs/RobotState.h>
+#include <moveit_msgs/PlannerInterfaceDescription.h>
+#include <moveit_msgs/Constraints.h>
+#include <moveit_msgs/Grasp.h>
+#include <moveit_msgs/PlaceLocation.h>
+#include <moveit_msgs/MotionPlanRequest.h>
+#include <moveit_msgs/MoveGroupAction.h>
+#include <moveit_msgs/PickupAction.h>
+#include <moveit_msgs/ExecuteTrajectoryAction.h>
+#include <moveit_msgs/PlaceAction.h>
+#include <geometry_msgs/PoseStamped.h>
+#include <moveit/robot_state/conversions.h>
+#include <actionlib/client/simple_action_client.h>
+#include <memory>
+#include <tf2_ros/buffer.h>
+
+namespace planning_scene_monitor
+{
+MOVEIT_CLASS_FORWARD(PlanningSceneMonitor);
+}
+
+namespace planning_pipeline
+{
+MOVEIT_CLASS_FORWARD(PlanningPipeline);
+}
+
+namespace plan_execution
+{
+MOVEIT_CLASS_FORWARD(PlanExecution);
+MOVEIT_CLASS_FORWARD(PlanWithSensing);
+}
+
+namespace trajectory_execution_manager
+{
+MOVEIT_CLASS_FORWARD(TrajectoryExecutionManager);
+}
+
+namespace
+{
+enum ActiveTargetType
+{
+  JOINT,
+  POSE,
+  POSITION,
+  ORIENTATION
+};
+}
+
+namespace moveit
+{
+/** \brief Simple interface to MoveIt! components */
+namespace planning_interface
+{
+class MoveItErrorCode : public moveit_msgs::MoveItErrorCodes
+{
+public:
+  MoveItErrorCode()
+  {
+    val = 0;
+  }
+  MoveItErrorCode(int code)
+  {
+    val = code;
+  }
+  MoveItErrorCode(const moveit_msgs::MoveItErrorCodes& code)
+  {
+    val = code.val;
+  }
+  explicit operator bool() const
+  {
+    return val == moveit_msgs::MoveItErrorCodes::SUCCESS;
+  }
+  bool operator==(const int c) const
+  {
+    return val == c;
+  }
+  bool operator!=(const int c) const
+  {
+    return val != c;
+  }
+};
+
+MOVEIT_CLASS_FORWARD(MoveItCpp);
+
+/** \class MoveItCpp move_group_interface.h moveit/planning_interface/move_group_interface.h
+
+    \brief Client class to conveniently use the ROS interfaces provided by the move_group node.
+
+    This class includes many default settings to make things easy to use. */
+class MoveItCpp
+{
+public:
+  /** \brief Default ROS parameter name from where to read the robot's URDF. Set to 'robot_description' */
+  static const std::string ROBOT_DESCRIPTION;
```

Comment:
using C++11, only define this in the cpp file using constexpr

Reply:
`ROBOT_DESCRIPTION` is used inside the header as default argument for the `Options` constructor. This will need more refactoring